src/appstatus.cpp

```diff
@@ -0,0 +1,171 @@
+#include "appstatus.h"
+
+AppStatus::AppStatus(QObject *parent)
+    : QObject(parent)
+    , m_status(Disabled)
+    , m_downloadStatus()
+    , m_uploadStatus()
+    , m_updating()
+    , m_bluetoothBusy()
+    , m_bluetoothEnabled()
+    , m_atRisk()
+    , m_downloadAvailable()
+{
+}
+
+AppStatus::Status AppStatus::status() const
+{
+    return m_status;
+}
+
+void AppStatus::updateStatus()
+{
+    Status status = m_status;
+
+    if (m_updating) {
+        m_status = Updating;
+    }
+    else if ((m_uploadStatus != Upload::StatusIdle) && (m_uploadStatus != Upload::StatusError)) {
+        m_status = Uploading;
+    }
+    else if ((m_downloadStatus != Download::StatusIdle) && (m_downloadStatus != Download::StatusError)) {
+        m_status = Downloading;
+    }
+    else if (m_uploadStatus == Upload::StatusError) {
+        m_status = ErrorUploading;
+    }
+    else if (m_downloadStatus == Download::StatusError) {
+        m_status = ErrorDownloading;
+    }
+    else if (m_bluetoothBusy) {
+        m_status = BluetoothBusy;
+    }
+    else if (m_atRisk) {
+        m_status = AtRisk;
+    }
+    else if (m_downloadAvailable) {
+        m_status = DailyUpdateRequired;
+    }
+    else if (m_bluetoothEnabled) {
+        m_status = Active;
+    }
+    else {
+        m_status = Disabled;
+    }
+
+    if (m_status != status) {
+        emit statusChanged();
+    }
+}
+
+Download::Status AppStatus::downloadStatus() const
+{
+    return m_downloadStatus;
```

Comment:
I think the function's name is misleading for outsiders. I would reccommend `AppStatus::statusOfDownload()`

Reply:
Thanks; I've changed them as you suggested.